src/components/panel/panel.js

```diff
@@ -932,6 +934,7 @@ MdPanelRef.prototype.detach = function() {
 
     self._panelContainer.remove();
     self.isAttached = false;
+    self._config.locals = null;
```

Comment:
@ErinCoughlan I'm all for a separate destroy method, I will change this to include that!